Removed Joda-Time dependency in framework and updated classes to use Java Date/Time

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4783

Removed Joda dependency in Pom's and fixed errors related to Joda time in all places

MarekB01 committed:
Removed Joda dependency in Pom's and fixed errors related to Joda time in all places

oleksiimiroshnyk commented on core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java:
```diff
@@ -497,17 +501,17 @@ protected List<Offer> removeOutOfDateOffers(List<Offer> offers){
-        DateTimeZone offerDateTimeZone = DateTimeZone.forTimeZone(offerTimeZone);
-        LocalDateTime offerDateTime = new LocalDateTime(date, offerDateTimeZone);
+        ZonedDateTime offerDateTimeZone = ZonedDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.of(String.valueOf(offerTimeZone)));
+        LocalDateTime offerDateTime = LocalDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.from(offerDateTimeZone));
```
maybe we need just LocalDateTime offerDateTime = LocalDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.of(String.valueOf(offerTimeZone))); ?
so we don't need offerDateTimeZone ?

riteshadhikari17 replied:
There are test failures which I think is because of casting here:
```
[ERROR] testApplyAllFulfillmentGroupOffersWithOrderItemOffers(org.broadleafcommerce.core.offer.service.processor.FulfillmentGroupOfferProcessorTest)  Time elapsed: 0.034 s  <<< ERROR!
java.lang.ClassCastException: java.util.Date cannot be cast to java.time.temporal.TemporalAccessor
	at org.broadleafcommerce.core.offer.service.processor.FulfillmentGroupOfferProcessorTest.testApplyAllFulfillmentGroupOffersWithOrderItemOffers(FulfillmentGroupOfferProcessorTest.java:275)
```

oleksiimiroshnyk commented on integration/src/test/java/org/broadleafcommerce/core/payment/service/NullIntegrationGatewayTransactionServiceImpl.java:
```diff
@@ -158,9 +159,9 @@ protected PaymentResponseDTO commonCreditCardProcessing(PaymentRequestDTO reques
-                    expirationDate = expirationDate.dayOfMonth().withMaximumValue();
-                    validDate = expirationDate.isAfterNow();
+                    Date expirationDate = new Date(Integer.parseInt("20" + expYear), Integer.parseInt(expMonth), 1, 0, 0);
+                    expirationDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
```
are you sure that you don't need to assign result of expirationDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();  ?

danielcolgrove commented:
@MarekB01 Changes seem good.  Did you do some thorough testing of offer start/end dates to make sure we didn't introduce any defects related to dates?

riteshadhikari17 commented on integration/src/test/java/org/broadleafcommerce/core/payment/service/NullIntegrationGatewayTransactionServiceImpl.java:
```diff
@@ -158,9 +160,9 @@ protected PaymentResponseDTO commonCreditCardProcessing(PaymentRequestDTO reques
-                    DateTime expirationDate = new DateTime(Integer.parseInt("20" + expYear), Integer.parseInt(expMonth), 1, 0, 0);
-                    expirationDate = expirationDate.dayOfMonth().withMaximumValue();
-                    validDate = expirationDate.isAfterNow();
+                    Date expirationDate = new Date(Integer.parseInt("20" + expYear), Integer.parseInt(expMonth), 1, 0, 0);
```
Looks like this is deprecated. Could you update this please?
![image](https://user-images.githubusercontent.com/35038090/195394187-455de61b-5957-436f-99d2-bd824228618c.png)

riteshadhikari17 commented:
@MarekB01 I rebased and squashed the commits. It seems to have fixed the issues. Could you test locally and verify?
@oleksiimiroshnyk Could you review this PR again please? There are new changes.

MarekB01 commented:
> @MarekB01 I rebased and squashed the commits. It seems to have fixed the issues. Could you test locally and verify? @oleksiimiroshnyk Could you review this PR again please? There are new changes.

My updates: tests passing after squashing a commits, build succeeded and verified locally

oleksiimiroshnyk commented on core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java:
```diff
@@ -497,17 +497,16 @@ protected List<Offer> removeOutOfDateOffers(List<Offer> offers){
+        LocalDateTime offerDateTime = LocalDateTime.ofInstant(date.toInstant(), offerTimeZone.toZoneId());
 
-        Calendar calendar = new GregorianCalendar(offerTimeZone);
+        Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone(String.valueOf(offerTimeZone)));
```
not sure we need to change here anything

oleksiimiroshnyk commented on integration/src/test/java/org/broadleafcommerce/core/payment/service/NullIntegrationGatewayTransactionServiceImpl.java:
```diff
@@ -157,10 +160,11 @@ protected PaymentResponseDTO commonCreditCardProcessing(PaymentRequestDTO reques
-                    expirationDate = expirationDate.dayOfMonth().withMaximumValue();
-                    validDate = expirationDate.isAfterNow();
+                    ZonedDateTime expirationDate = ZonedDateTime.of((Integer.parseInt("20" + expYear)), Integer.parseInt(expMonth), 1, 0, 0, 0, 0, zone);
+                    LocalDateTime expDate = expirationDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
```
probably can do
LocalDateTime expDate = expirationDate.toInstant().atZone(zone).toLocalDateTime();

riteshadhikari17 committed:
Make changes requested during after code review
